src/config/entities.ts

```diff
@@ -87,24 +88,24 @@ export interface CategoryConfig {
 
 /**
  * Column configuration.
+ * TanStack ColumnDef properties not currently supported include:
+ * - `enableMultiSort` - Note, table multi-sort is managed via table options.
  */
-export interface ColumnConfig<
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- This config model is part of a generic array
-  T = any,
+export type ColumnConfig<
+  T extends RowData,
+  TValue = unknown,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any -- This config model is part of a generic array
   C extends keyof JSX.IntrinsicElements | JSXElementConstructor<any> = any
-> {
+> = Omit<ColumnDef<T, TValue>, "enableMultiSort"> & {
```

Comment:
Have omitted multi-sort on the actual column definition to reduce complexity in the toggling of sorting etc!
Extended ColumnConfig with TanStack `ColumnDef` - we want to better control over what a column can and can't do!! Ultimate goal here is to only use ColumnDef.